### DISCUSSION TITLE
Unistall

### DISCUSSION THREAD

**Discussion by cclavijo** [AdityaGarg8/mbp-16.x-ubuntu-kernel#32] [Q&A] [answered]
Which should be the steps to rollback the installation?

**Comment by AdityaGarg8** [comment]
> Which should be the steps to rollback the installation?

You want to rollback the installation of the kernel or complete Ubuntu installation?

**Comment by cclavijo** [comment]
roll back the installation of the kernel. :)

**Reply by RishonDev** [reply]
I have given an detailed answer :)

**Comment by RishonDev** [comment]
- If you are trying to remove the whole OS, then:
1) If you don't have macOS, then shutdown the computer. then Start up and before the the logo shows up and the computer is on, Hold Command + Option + R for 10-20 seconds. 

Choose your network and type your password. Now it will download macOS recovery which is a few gigabytes, so be patient. After it finishes, Erase the whole disk using disk utility(If it doesn't show the whole disk, Click on View > Show all devices).

 Quit disk utility  and click on the installer. Follow the installation instructions and let it do its thing. once it finishes it should say it's going to reboot. click on the restart button and go through the one-time setup and you are done.


2) If you have macOS installed(dual booted with linux) Then:
Boot into your linux system and remove the EFI partition.
Reboot into you macOS System. Open Disk Utility and delete the partitions.

And you are done.

- If you are trying to rollback the kernel, Then:

Open terminal. Type this:
```bash
curl -s --compressed "https://adityagarg8.github.io/t2-ubuntu-repo/KEY.gpg" | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/t2-ubuntu-repo.gpg >/dev/null
sudo curl -s --compressed -o /etc/apt/sources.list.d/t2.list "https://adityagarg8.github.io/t2-ubuntu-repo/t2.list"
sudo apt update
```

If you want an LTS version of the kernel, then:
```
sudo apt install t2-kernel-script-lts
```
or else 
```
sudo apt install t2-kernel-script
```

Then copy and paste the command
```
 update_t2_kernel --remove-current
```

**Comment by AdityaGarg8** [comment]
> Which should be the steps to rollback the installation?

Alright.

Can you tell me the exact reason you wanna do that.
Secondly, you wanna roll back to a minor version (like from 6.0.11 to 6.0.10) , or a major release (like from 6.0.11 to 5.15.81)

**Reply by cclavijo** [reply]
I think i not explain it to much. I installed t2-kernel-script, and now i want to rollback (like unistall the 5.15.82-t2-kinetic) and install normal linux one. can we do that?

**Comment by AdityaGarg8** [comment, marked as answer]
Run :-

```
sudo apt remove linux-headers-5.15.82-kinetic linux-image-5.15.82-kinetic t2-kernel-script
```

**Reply by cclavijo** [reply]
ok, but niw i dont have any kernel installed if i restart my computer will not boot rigth?
